### DISCUSSION TITLE
Regarding watch channel guarantee and broadcast channel drop value condition.

### DISCUSSION THREAD

**Discussion by realzhujunhao** [tokio-rs/tokio#5948] [Q&A]
Hello, I'm new to tokio async, currently struggling on which channel to use.

I have one producer task and multiple consumer task, each consumer task is going to block (or yield? I'm not sure about correct term) by polling the receiver until its desired value is given. `watch` and `broadcast` channels seem to be suitable for my use case, and since I have only one producer, `watch` might be the best fit.

For example, suppose my producer task send random chars, consumer task 1 wait until it sees 'A' to proceed the rest of its logic (other chars are ignored but I still want them visible to other consumers).

My concerns with `watch`:
I've read through the documentation, it says "A single-producer, multi-consumer channel that only retains the last sent value". Is it possible that a consumer task is not scheduled to run in the time interval producer sends two value? i.e Some consumer task might miss some value.

My concerns with `broadcast`:
Doc says "messages must be retained until all handles receive a clone", in my case the desired value for one consumer is not useful for other consumers. 

In summary,
1. Which one of `watch` and `broadcast` should I use?
2. I am also considering to use mpsc to send `oneshot Sender` from consumer tasks to producer task, but this will make my code very complicated and I'm not even sure if that works or not.

**Comment by Darksonn** [comment]
* `watch` Yes, a receiver may miss a value if it's too slow.
 * `broadcast` When a broadcast channel fills up, messages are dropped and receivers are told how many values were skipped. If skipping values is not okay for you, then you don't want this one either.

You may want to try a broadcast channel that doesn't skip values when it fills up, there should be crates out there with such implementations. Alternatively, you could use many mpsc channels, one for each receiver.
